### ISSUE TITLE
Only allow moloch members to submit proposals

### ISSUE THREAD

**Issue by proofoftom** [raid-guild/moloch-minion#28]
Currently anyone can submit a proposal which could lead to an adversary submitting many junk proposals making it impossible to filter fact from fiction. To rectify, only members of the respective moloch's minion should be able to submit proposals with that minion.

**Comment by wolflo** [comment]
I think this is a good discussion to have. The complexity of figuring out what a minion proposal does means it potentially deserves more barriers than a standard proposal. On the other hand, the molochV2 ideal seems to be that proposal submission is free and open and sponsorship is what separates spam from valid proposals.

**Comment by dekanbro** [comment]
yeah was thinking we should allow a minion submitter to add tribute. Adding a small tribute would be like a fee paid to the dao for the extra audit resposibilities. even staking a small fee would probably weed out most spam.

**Comment by wolflo** [comment]
Yeah that's a good point. Are you thinking a tribute to the dao or directly to the minion? If it's a deposit, I suppose it would have to just be managed by the minion. But if its a permanent tribute, we could give it to the minion or have the minion pass it on to the moloch.

I think if it's a payment rather than a deposit, it might discourage a lot of use cases that require extra executions (e.g. ERC20 approves), unless we also expanded the current one-execution-at-a-time method.

**Comment by wolflo** [comment]
If we were going to implement some sort of "proposal control", would be interesting to look into something along the lines of [this](https://github.com/dapphub/ds-guard/blob/4678e1c74fce1542f1379f11325d7bfbbb897344/src/guard.sol#L47-L55) to allow daos to set their own standards for who can submit a proposal. Maybe opening it to anyone, limiting it to members, or setting a whitelist.

Would also require setting a standard for some sort of dao-to-minion communication, though. We are somewhat limited by how much important information it's reasonable to encode in the details string or embed in the minion itself. The more complexity we add, the more difficult it gets to decipher what a proposal will do. Oh, and in the details string we're also limited to utf-8 data if we want external libraries to work at the moment.
